storage/local/persistence.go

```diff
@@ -606,7 +608,9 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			if err != nil {
 				return
 			}
-			w.Write(buf)
+			if _, err = w.Write(buf); err != nil {
```

Comment:
Yes, error logging in https://github.com/prometheus/prometheus/blob/master/storage/local/storage.go#L878 (and not reset the dirty series count... which is a good example why the error should be returned and not just logged within the function).